lib/prepare-write.js

```diff
@@ -5,39 +5,34 @@ var path = require('path');
 var mkdirp = require('mkdirp');
 var fs = require('graceful-fs');
 
-function booleanOrFunc(v, file) {
-  if (typeof v !== 'boolean' && typeof v !== 'function') {
-    return null;
-  }
+var valueOrFunction = require('value-or-function');
+var defaultValue = require('./default-value');
 
-  return typeof v === 'boolean' ? v : v(file);
-}
+var boolean = valueOrFunction.boolean;
+var number = valueOrFunction.number;
+var string = valueOrFunction.string;
 
-function stringOrFunc(v, file) {
-  if (typeof v !== 'string' && typeof v !== 'function') {
-    return null;
+function prepareWrite(outFolder, file, opt, cb) {
+  if (!opt) {
+    opt = {};
   }
 
-  return typeof v === 'string' ? v : v(file);
-}
-
-function prepareWrite(outFolder, file, opt, cb) {
-  var options = assign({
-    cwd: process.cwd(),
-    mode: (file.stat ? file.stat.mode : null),
-    dirMode: null,
-    overwrite: true,
-  }, opt);
-  var overwrite = booleanOrFunc(options.overwrite, file);
-  options.flag = (overwrite ? 'w' : 'wx');
+  var defaultMode = file.stat ? file.stat.mode : null;
+  var options = assign({}, opt, {
```

Comment:
Two reasons:
- it was done like that before I came in
- namespacing

Would you prefer individual vars suffixed with `Opt`? Same in `src/index.js`? Is that for performance reasons?